backend/logic/courses/courses.js

```diff
@@ -3,23 +3,31 @@ const format = require("../../validation/format");
 
 /**
  * Returns a list of courses matching the tag
- * @author Alex Lam
+ * @author Alex Lam, Feras Al Taha
  * @param {string} tag
  * @returns A list of courses in JSON format
  * @throws Undefined tag if tag is null
  *         error if MySQL connection failed
  */
-var getCourseByTag = async function getCourseByTag(tag) {
+var getCourseByTag = async function getCourseByTag(tag,studentID) {
```

Comment:
Isn't this endpoint used to return courses matching a tag for everyone and not for a specific student?

Reply:
Well the page displaying the courses by tag also needed to show if those courses were marked as desired or not so we decided to modify this endpoint so that it would still return the same courses as before but with an additional parameter specifying if they were previously marked as desired or not by the student